### ISSUE TITLE
Action to input unicode character

### ISSUE THREAD

**Issue by frog-o** [keymapperorg/KeyMapper#165]
I want to type be able to type special Unicode Character such as ñ on my Bluetooth keyboard which you do not include in your key maps  it would be nice if you would allow type there charter code in.

**Comment by sds100** [comment]
Yeah, that's a great idea. I'll add a list to the Choose Action activity which displays all the unicode characters.

**Comment by sds100** [comment]
https://developer.android.com/reference/android/icu/text/UnicodeSet get unicode characters from here?

**Comment by Wilker-a** [comment]
[UnicodePad](/Ryosuke839/UnicodePad) is a useful tool for getting unicode characters. maybe figuring a way to gather info from the app (or simply suggesting to install it on the application) might save some work?

**Comment by sds100** [comment]
Thanks for the suggestion. I'll look at how they get unicode characters to have an idea for how to do it for mine. I'll give them credit if I use the same method.

**Comment by sds100** [comment]
- Use xml files from here: https://www.unicode.org/Public/UCD/latest/ to get a list of all unicode chars
- Shrink the xml file to only include necessary data. Use DB if too slow to deserialize??

**Comment by sds100** [comment]
@frog-o Couldnt you use the "input text" action?

**Comment by mjibe** [comment]
Hello,

I confirm this feature would be very useful, as we could also link to diacritical characters to have native android behaviour for dead keys [(android doc)](https://source.android.com/devices/input/key-character-map-files#behaviors)
For example (my sumsung tab S5e lacks some keys on Cover keyboard, I remapped most of them successfully, but I'm stuck with dead keys, which prevents me to type some accentuated characters natively).

**Comment by sds100** [comment]
> Hello,
> 
> I confirm this feature would be very useful, as we could also link to diacritical characters to have native android behaviour for dead keys [(android doc)](https://source.android.com/devices/input/key-character-map-files#behaviors) For example my sumsung tab S5e lacks some keys on Cover keyboard.
> 
> Issue (on samsung side): every accessibility service targeting input messes with physical charmaps. Then, I remapped most of them successfully, but I'm stuck with dead keys, which prevents me to type some accentuated characters natively).
> 
> Even if there's not a list, it would be useful to have "unicode character" action (as there's plenty of unicode charlist outside).

@mjibe That issue sounds like this bug in Android 11. I have made a work around for it, which you can read about here https://docs.keymapper.club/known-issues/#external-keyboard-language-is-set-to-english-us-when-using-key-mapper-android-11-only

You should just be able to use the "input text" action if you want to input a unicode character.
